lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -668,6 +679,7 @@ postWallet
         , IsOurs s RewardAccount
         , Typeable s
         , Typeable n
+        , (k == SharedKey) ~ 'False
```

Comment:
I am a bit confused by this.

Reply:
so we use this here https://hackage.haskell.org/package/base-4.15.0.0/docs/src/Data-Type-Equality.html#%3D%3D and we needed as we introduced different ParentContext for SharedKey and the rest (ShelleyKey/IcarusKey..). So we want here to say that k must not be SharedKey. for SharedKey we just specify `k ~ SharedKey`